src/Utility/Models/Criteriable.php

```diff
@@ -53,43 +58,52 @@ public function scopeFilterByCriteria($query, array $criteria): Builder
 
     private function buildMatchAgainst($query, array $columns, $search)
     {
-        return $query->whereRaw("MATCH(" . implode(',', $columns) . ") AGAINST (? " . $this->fullTextSearchMode . ")", $search);
+        return $query->whereRaw('MATCH('.implode(',', $columns).') AGAINST (? '.$this->fullTextSearchMode.')', $search);
     }
 
-    public function scopeSearchByCriteria($query, array $criteria): Builder
+    /**
+     * @param callable(Builder, array):Builder|null $relevance
+     */
+    public function scopeSearchByCriteria($query, array $criteria, callable $relevance = null): Builder
     {
-        if (isset($criteria['search'])) {
-            $query->where(function ($q) use ($criteria) {
-                foreach ($this->fulltextSearch as $method => $columns) {
-                    if (method_exists($this, $method)) {
-                        $q->orWhereHas($method, function ($query) use ($criteria, $columns) {
-                            $this->buildMatchAgainst($query, $columns, $criteria['search']);
-                        });
-                    } else {
-                        $this->buildMatchAgainst($q, $columns, $criteria['search']);
-                    }
+        if (!isset($criteria['search'])) {
+            return $query;
+        }
+
+        if($relevance && is_callable($relevance)) {
+            $relevance($query, $criteria);
```

Comment:
sorry but this doesn't make sense, the `searchByCriteria` method is already called in the builder, in the developer needs to add more query he can add them on the builder chain, they don't need to add a query as a parameter in this method. am I missing something?

Reply:
Then in that case there's no need for changes to make the relevance work inside the package, just let me update the docs